src/metabase/driver/druid.clj

```diff
@@ -70,17 +70,31 @@
 
 ;;; ### Sync
 
-(defn- describe-table-field [druid-field-type field-name]
+(defn- do-segment-metadata-query [details datasource]
+  (do-query details {"queryType"     "segmentMetadata"
+                     "dataSource"    datasource
+                     "intervals"     ["1999-01-01/2114-01-01"]
+                     "analysisTypes" []
+                     "merge"         true}))
+
+(defn- druid-type->base-type [field-type]
+  (case field-type
+    "STRING"      :type/Text
+    "FLOAT"       :type/Float
+    "LONG"        :type/Float
+    "hyperUnique" :type/DruidHyperUnique
+    :type/Float))
+
+(defn- describe-table-field [field-name {field-type :type, :as info}]
   ;; all dimensions are Strings, and all metrics as JS Numbers, I think (?)
   ;; string-encoded booleans + dates are treated as strings (!)
-  (assoc (case druid-field-type
-           :metric    {:database-type "metric",    :base-type :type/Float}
-           :dimension {:database-type "dimension", :base-type :type/Text})
-    :name field-name))
+  {:name          (name field-name)
```

Comment:
Nit picky, but seems like `name` here is not needed.

Reply:
turns out it is needed :)